selfhost/lexer.jakt

```diff
@@ -370,6 +442,10 @@ struct Lexer implements(ThrowingIterable<Token>) {
                 .error("reserved identifier name", span)
             }
 
+            if is_illegal_cpp_keyword(string) {
+                .error("c++ keyword", span)
```

Comment:
Let's use a more descriptive error string here.

Reply:
"c++ keyword used as identifier"?

Reply:
An actual error saying what's wrong would help, e.g. "C++ keywords are not allowed to be used as identifiers"